### PR TITLE
Add application name to DB

### PR DESCRIPTION
**Pull request by zackkrida**
Adds our application name to the database options. This can be useful for identifying the source of queries in PostgreSQL's `pg_stat_activity` view or in query logs.

### CODE REVIEW

**Review by krysal — APPROVED** [review]

**Review by AetherUnbound — APPROVED** [review]
LGTM

File: api/catalog/settings.py [hunk 1]
```diff
@@ -260,6 +260,9 @@ def _make_cache_config(dbnum: int, **overrides) -> dict:
         "USER": config("DJANGO_DATABASE_USER", default="deploy"),
         "PASSWORD": config("DJANGO_DATABASE_PASSWORD", default="deploy"),
         "NAME": config("DJANGO_DATABASE_NAME", default="openledger"),
+        "OPTIONS": {
+            config("DJANGO_DATABASE_APPLICATION_NAME", default="openverse-api"),
+        },
```
Comment (sarayourfriend) [inline api/catalog/settings.py:265]: Is `OPTIONS` really meant to be a set? Is a key missing?

**Review by sarayourfriend — CHANGES_REQUESTED** [review]

File: api/catalog/settings.py (hunk 1 above)
Comment (AetherUnbound) [reply api/catalog/settings.py:265]: Oh good catch

Comment (zackkrida) [reply api/catalog/settings.py]: GAH, fixed. Thank you.

Comment (krysal) [reply api/catalog/settings.py]: It's a [dictionary](https://docs.djangoproject.com/en/4.1/ref/settings/#std-setting-OPTIONS) but I can't more specific information for Postgres yet.

**Review by AetherUnbound — APPROVED** [review]

**Review by krysal — APPROVED** [review]

**Pull request merged**